### PR TITLE
*: exit process when panic.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @BusyJay

### CODE REVIEW

File: src/util/panic_hook.rs [hunk 1]
```diff
@@ -71,3 +72,12 @@ fn track_hook(p: &PanicInfo) {
         }
     });
 }
+
+/// Exit the whole process when panic.
+pub fn set_exit_hook() {
+    let orig_hook = panic::take_hook();
+    panic::set_hook(box move |info: &PanicInfo| {
+        orig_hook(info);
+        process::exit(1);
```
Comment (ngaut) [inline src/util/panic_hook.rs]: Do we need to print the panic info before exit process?

Comment (siddontang) [inline src/util/panic_hook.rs]: The default origin hook will do this.

Comment (BusyJay) [inline src/util/panic_hook.rs]: Is it safe to just exit? Should we do it gracefully?

Comment (siddontang) [inline src/util/panic_hook.rs]: no need.
we must guarantee safety in our code, not in panic.

**Comment by BusyJay** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**